src/range.rs

```diff
@@ -33,7 +33,7 @@ macro_rules! def_range {
             pub const fn from_start_size(start: $addr_type, size: usize) -> Self {
                 Self {
                     start,
-                    end: <$addr_type>::from(usize!(start) + size),
+                    end: <$addr_type>::from_usize(usize!(start) + size),
```

Comment:
Keep `from` to minimize API changes.

Reply:
It somewhat conflicts with `From::<usize>::from` so (even Rust allows it) I think it would be better to give it another name.

It won't be a break change because `PhysAddr/VirtAddr::from` still compiles now, with the same signature and the same behavior. The only problem is `const` but I think it's just acceptable (as `va!` and `pa!` are still there).

Reply:
I share the same concern.

when you rename `from` to `from_usize` and left `from` method in `MemoryAddr` trait, in every place that use `PhysAddr::from` or `VirtAddr::from` will have to import the `MemoryAddr` trait, it will be a large lot of changes.

Reply:
BUT, from_usize seems more sensible.

Reply:
> I share the same concern.
> 
> 
> 
> when you rename `from` to `from_usize` and left `from` method in `MemoryAddr` trait, in every place that use `PhysAddr::from` or `VirtAddr::from` will have to import the `MemoryAddr` trait, it will be a large lot of changes.

Haven't tested it yet but I don't think users will be forced to `use MemoryAddr`, as `From` is already precluded.

Reply:
> Haven't tested it

Tested now.

Reply:
Can you evaluate the effort of crate upgrading in ArceOS?

Reply:
> Can you evaluate the effort of crate upgrading in ArceOS?

I'll evaluate it tonight and tomorrow.